app/scripts/services/secrets.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("SecretsService", function($q, $filter, DataService){
+
+    var newestSecret;
+
+    var setNewestSecretName = function(secretName) {
+      newestSecret = secretName;
+    };
+
+    var getNewestSecretName = function() {
+       return newestSecret;
+    };
+
+    var loadSecrets = function(projectName, $scope) {
```

Comment:
The pattern that we've taken elsewhere it to separate the list from the grouping methods. So I'd expect this function to take in a list of secrets and group them (`var groupSecretsByType = function ...`)

Generally the controllers manage the calls to DataService when possible.